sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -202,7 +211,7 @@ object FileFormatWriter extends Logging {
             committer,
             iterator = iter)
         },
-        0 until rdd.partitions.length,
+        0 until rddWithNonEmptyPartitions.partitions.length,
```

Comment:
nit: `rddWithNonEmptyPartitions.partitions.indices`

Reply:
made the change.  Learnt a new thing today :-)